app/components/search_bar/search_bar.ios.js

```diff
@@ -1,96 +1,145 @@
     static propTypes = {
-        barStyle: PropTypes.oneOf(['default', 'black']),
-        searchBarStyle: PropTypes.oneOf(['default', 'prominent', 'minimal']),
-        text: PropTypes.string,
-        placeholder: PropTypes.string,
-        showCancelButton: PropTypes.bool,
-        hideBackground: PropTypes.bool,
-        textFieldBackgroundColor: PropTypes.string,
-        placeholderTextColor: PropTypes.string,
-        textColor: PropTypes.string,
-        barTintColor: PropTypes.string, //color of the background container
-        tintColor: PropTypes.string, // color of the carret and the cancel button
-        onChange: PropTypes.func,
+        onCancelButtonPress: PropTypes.func,
         onChangeText: PropTypes.func,
         onFocus: PropTypes.func,
-        onBlur: PropTypes.func,
         onSearchButtonPress: PropTypes.func,
-        onCancelButtonPress: PropTypes.func
-    };
-
-    static defaultProps = {
-        barStyle: 'default',
-        searchBarStyle: 'minimal',
-        placeholder: 'Search',
-        showCancelButton: true,
-        hideBackground: true,
-        onFocus: () => true,
-        onBlur: () => true
+        backgroundColor: PropTypes.string,
+        placeholderTextColor: PropTypes.string,
+        titleCancelColor: PropTypes.string,
+        tintColorSearch: PropTypes.string,
+        tintColorDelete: PropTypes.string,
+        cancelButtonStyle: PropTypes.PropTypes.oneOfType([
+            PropTypes.number,
+            PropTypes.object
+        ]),
+        inputStyle: PropTypes.oneOfType([
+            PropTypes.number,
+            PropTypes.object
+        ]),
+        placeholder: PropTypes.string,
+        cancelTitle: PropTypes.oneOfType([
+            PropTypes.string,
+            PropTypes.object
+        ]),
+        returnKeyType: PropTypes.string,
+        keyboardType: PropTypes.string,
+        autoCapitalize: PropTypes.string,
+        inputHeight: PropTypes.number,
+        inputBorderRadius: PropTypes.number,
+        blurOnSubmit: PropTypes.bool
     };
 
     constructor(props) {
         super(props);
 
         this.state = {
-            displayCancelButton: false
+            placeholderWidth: 0
         };
     }
 
-    onFocus = (event) => {
-        const onFocus = this.props.onFocus;
-        if (this.props.showCancelButton) {
-            this.setState({displayCancelButton: true});
-        }
+    shouldComponentUpdate(nextProps, nextState) {
+        return nextState.placeholderWidth !== this.state.placeholderWidth;
+    }
 
-        onFocus(event);
+    afterDelete = () => {
+        return new Promise((resolve) => {
+            this.refs.search.focus();
+            resolve();
+        });
     };
 
-    onBlur = (event) => {
-        const onBlur = this.props.onBlur;
-        if (this.props.showCancelButton) {
-            this.setState({displayCancelButton: false});
-        }
+    cancel = () => {
+        this.refs.search.onCancel();
+    };
+
+    onCancel = () => {
```

Comment:
because the library that we are using requires them to be promises https://github.com/crabstudio/react-native-search-box